typing/includemod.ml

```diff
@@ -480,17 +498,19 @@ let _ = Env.check_modtype_inclusion := check_modtype_inclusion
 
 let compunit env impl_name impl_sig intf_name intf_sig =
   try
-    signatures env [] Subst.identity impl_sig intf_sig
+    signatures ~loc:(Location.in_file impl_name) env [] Subst.identity
+      impl_sig intf_sig
   with Error reasons ->
     raise(Error(([], Env.empty,Interface_mismatch(impl_name, intf_name))
                 :: reasons))
 
 (* Hide the context and substitution parameters to the outside world *)
 
-let modtypes env mty1 mty2 = modtypes env [] Subst.identity mty1 mty2
-let signatures env sig1 sig2 = signatures env [] Subst.identity sig1 sig2
-let type_declarations env id decl1 decl2 =
-  type_declarations env [] Subst.identity id decl1 decl2
+let modtypes ~loc env mty1 mty2 = modtypes ~loc env [] Subst.identity mty1 mty2
+let signatures env sig1 sig2 =
```

Comment:
Includemod.signatures is only used in context which could not trigger the warning, I believe (in {opt,}{compile,toploop}.ml).  Moreover the resulting coercion is ignored in those cases.  Perhaps it would be better to expose a more specific function (better name and a `unit` result) to cover those cases (do you happen to remember why we need to check self-inclusion of signatures e.g. in compiler.ml?).  But this is not really related to this PR anyway.